packages/itwin/tree-widget/src/test/trees/stateless/models-tree/internal/ModelsTreeVisibilityHandler.test.ts

```diff
@@ -0,0 +1,2266 @@
+              const props: GeometricElement3dProps = {
+                classFullName: "Generic:PhysicalObject",
+                code: new Code({ scope: partitionId, spec: "", value: `${categoryIdx}_${className}` }),
+                model: modelId,
+                category: categoryId,
+              };
+              const parent = builder.insertElement(props);
+              parents.push(parent);
+              const elements = new Set<string>();
+              for (let childIdx = 0; childIdx < 2; ++childIdx) {
+                const element = builder.insertElement({
+                  ...props,
+                  code: new Code({ scope: partitionId, spec: "", value: `${categoryIdx}_${className}_${childIdx}` }),
+                  classFullName: `test:${className}`,
+                  parent: { id: parent, relClassName: "BisCore:PhysicalElementAssemblesElements" },
+                });
+                elements.add(element);
+              }
+              elementHierarchy.set(parent, elements);
+            }
+            categoryParentElements.set(categoryId, parents);
+          }
+          models.set(modelId, modelCategories);
+        }
+
+        [firstModelId, otherModelId] = [...models.keys()];
+      });
+
+      iModelConnection = await SnapshotConnection.openFile(iModelPath);
+    });
+
+    after(async () => {
+      await iModelConnection?.close();
+      iModel?.close();
+      iModelPath && fs.rmSync(iModelPath);
+      await terminatePresentationTesting();
+    });
+
+    let viewport: Viewport;
+    let handler: ModelsTreeVisibilityHandler;
+
+    beforeEach(async () => {
+      viewport = OffScreenViewport.create({
+        view: createBlankViewState(iModelConnection),
+        viewRect: new ViewRect(),
+      });
+      handler = createModelsTreeVisibilityHandler({
+        viewport,
+      });
+    });
+
+    afterEach(() => {
+      handler.dispose();
+    });
+
+    function filterMatches(id: string, idsFilter: undefined | string | ((id: string) => boolean)) {
+      if (!idsFilter) {
+        return true;
+      }
+      if (typeof idsFilter === "string") {
+        return id === idsFilter;
+      }
+      return idsFilter(id);
+    }
+
+    async function assertModelVisibility(props: { viewportVisibility: boolean; handlerVisibility: Visibility; modelId?: Id64String }) {
+      const modelIds = props.modelId ? [props.modelId] : [...models.keys()];
+      await Promise.all(
+        modelIds.map(async (id) => {
+          expect(viewport.view.viewsModel(id)).to.eq(props.viewportVisibility, `Model ${id} has unexpected viewport visibility`);
+          const status = await handler.getVisibilityStatus(createModelHierarchyNode(id));
+          expect(status.state).to.eq(props.handlerVisibility, `Model ${id} has unexpected visibility in the handler`);
+        }),
+      );
+    }
+
+    async function assertCategoryVisibility(props: {
+      viewportVisibility?: boolean;
+      perModelVisibilityOverride?: boolean;
```

Comment:
can we change this to `"none" | "show" | "hide"`?

Reply:
Changed